### PR TITLE
[DOCS] add missing mention of current version to snapshot docs

### PR DESCRIPTION
**Pull request by geekpete**
It's come up a couple of times that the current docs for snapshot compatibility
don't mention the current versions.
There's likely a reworking of this page to come but just plugging this gap
to reduce any confusion or doubt around version compatibility in the meantime.
Requires backport to all 7.x branches.

### CODE REVIEW

File: docs/reference/modules/snapshots.asciidoc [hunk 1]
```diff
@@ -46,12 +46,15 @@ A snapshot contains a copy of the on-disk data structures that make up an
 index. This means that snapshots can only be restored to versions of
 Elasticsearch that can read the indices:
 
+* A snapshot of an index created in 6.x can be restored to 7.x.
 * A snapshot of an index created in 5.x can be restored to 6.x.
 * A snapshot of an index created in 2.x can be restored to 5.x.
 * A snapshot of an index created in 1.x can be restored to 2.x.
 
 Conversely, snapshots of indices created in 1.x **cannot** be restored to 5.x
 or 6.x, and snapshots of indices created in 2.x **cannot** be restored to 6.x.
```
Comment (jrodewig) [inline docs/reference/modules/snapshots.asciidoc]:
```suggestion
or 6.x, snapshots of indices created in 2.x **cannot** be restored to 6.x
```

**Review by jrodewig — CHANGES_REQUESTED** [review]
Left some needed grammatical corrections.

I also think this convention is getting a little unwieldy, but that's a separate problem from including the latest info. Created #46596 to address the information design issue.

Thanks @geekpete!

**Review by jrodewig — APPROVED** [review]
LGTM. Thanks.

I'll merge and backport this.

**Comment by colings86** [comment]
It looks like the backporting is complete here so I have removed the backport pending label while generating 7.4.0 release notes. Please let me know if this is wrong

**Pull request merged**